doc/notes/asl2masl_howto.adoc

```diff
@@ -0,0 +1,228 @@
+activity signature.  The parser will parse the MASL signatures and then
+switch to ASL mode and parse the ASL action bodies.  And abstract syntax
+tree results which is then "walked" in a separate step to populate the OOA
+of ASL/MASL Activity.
+
+=== 2.4 Textual MASL Output
+
+Textual MASL is rendered from the populated OOA of ASL/MASL.  This is
+accomplished using string transformations on the metamodel elements.  A
+recursive descent/ascent approach is used navigating from the top level of
+an activity down through blocks and statement to the lowest "leaf" levels
+of expressions and literal values.
+
+== 3 HOWTO
+
+=== 3.1 HOWTO Mark the Domain Version
+
+The MASL Exporter will look for a 'version' mark on the domain component
+of the model being converted.  If this mark does not exist, then required
+pragmas for ASL-to-MASL conversion will be missing.  Therefore, mark the
+domain with a version value.  The specific value of the version is not
+used, only its existence matters.
+
+Use the Marking Editor ("Manage Project Markings" from the project context
+menu in BridgePoint).  Mark the domain Component with a version of any value.
+
+=== 3.2 HOWTO Obtain a Marked `.mod` File
+
+To obtain a marked `.mod` file and all of the ASL activities, run the MASL
+Exporter (in MASL mode) from the command line.
+
+ <path to BridgePoint installation>/tools/mc/bin/xtuml2masl -xf -i <path to xtUML project> -d <domain name> -o <output folder>
+
+where:
+
+* <path to BridgePoint installation> is the filesystem path to a local
+  installation of BridgePoint from July 2023 or newer.
+* <path to xtUML project> is a path to an Eclipse/BridgePoint project
+  folder.
+* <domain name> is the name of the domain component in the model.
+* <output folder> is a folder into which exported MASL will be placed.
+
+An example invocation follows:
+
+ ~/xtuml/BridgePoint/tools/mc/bin/xtuml2masl -xf -i . -d Events -o outdir
+
+=== 3.3 HOWTO Run ASL2MASL
+
+The ASL2MASL utility is packaged as a Java `.jar` file.  The command line
+usage looks like this:
+
+ java -jar <path to jar file>/asl2masl.jar --mod <path to .mod file> --asl
+
+* 'java' will invoke your locally install JVM.
+* 'asl2masl.jar' is the packaged ASL2MASL program.
+* '--mod <path to .mod file>' directs the program to the `.mod` file and
+  expects all of the ASL activities to be sibling files in the same
+  folder.
+* '--asl' is a switch to indicate that ASL idiom models are the input.
+
+Here is an example invocation:
+
+ java -jar target/asl2masl.jar --mod ~/git/models/masl/SWATS/Struct/Struct.mod --asl
+
+=== 3.4 HOWTO Run the MASL C++ Generator
+
+After translating ASL to MASL, run the MASL C++ code generator to detect
+errors (and shortcomings) in the resulting MASL activity source.  The
+`.mod` file must be present (copied over) to get syntax-checking results.
+Usage of the MASL translator is as follow:
+
+  <path to MASL C++ translator>/masl-core -domainpath <path to MASL source> -mod <path to .mod file>
```

Comment:
-domainpath is not necessary for the main .mod file. It is only necessary if the model refers to other (external) domains like utility domains.